sdk/src/main/java/com/google/cloud/dataflow/sdk/util/BigQueryServices.java

```diff
@@ -93,6 +93,13 @@ Job pollJob(JobReference jobRef, int maxAttempts)
      */
     JobStatistics dryRunQuery(String projectId, String query)
         throws InterruptedException, IOException;
+
+    /**
+     * Check existence of a job.
+     *
+     * <p>Returns null if the {@code maxAttempts} retries reached.
+     */
+    Boolean exists(JobReference jobRef, int maxAttempts) throws InterruptedException;
```

Comment:
I suggests to define the new function as:
/**
* Gets the Job by {@link JobReference}.
* <p>Returns null if the job is not found.
*/
Job getJob(JobReference jobRef) throws InterruptedException, IOException;

The reason is: 1. getJob doesn't need to wait for the job to finish. 2. getJob is more general than exists.
Use the default BackOff for IOException:
new AttemptBoundedExponentialBackOff(MAX_RPC_ATTEMPTS, INITIAL_RPC_BACKOFF_MILLIS)

And, In the CleanupOperation, you can replace the pollJob() with getJob().